### PR TITLE
feat: implement createNewTask in ServiceManager and send tasks using CLI

### LINKED ISSUE #19: feat: implement createNewTask in ServiceManager
- Implement createNewTask method in ServiceManager Smart Contract
- Allow Task Sender CLI to interact with the new method

### PR DESCRIPTION
**Pull request by JuArce**
feat: create task using CLI
- using command `make send-plonk-proof` interact with the ServiceManager creating a dummy task

feat: add createNewTask method in ServiceManager
- createNewTask emits a `NewTaskCreated` event

chore: update anvil states

chore: update bindings

### CODE REVIEW

File: core/chainio/avs_writer.go [hunk 1]
```diff
@@ -82,26 +86,58 @@ func NewAvsWriterFromConfig() (*AvsWriter, error) {
...
-// 	receipt := w.client.WaitForTransactionReceipt(ctx, tx.Hash())
-// 	newTaskCreatedEvent, err := w.AvsContractBindings.TaskManager.ContractAlignedLayerTaskManagerFilterer.ParseNewTaskCreated(*receipt.Logs[0])
-// 	if err != nil {
-// 		w.logger.Error("Aggregator failed to parse new task created event", "err", err)
-// 		return cstaskmanager.IAlignedLayerTaskManagerTask{}, 0, err
-// 	}
-// 	return newTaskCreatedEvent.Task, newTaskCreatedEvent.TaskIndex, nil
-// }
+//
+//	func (w *AvsWriter) SendNewTaskVerifyProof(ctx context.Context, proof []byte, pubInput []byte, verifierId common.VerifierId, quorumThresholdPercentage uint32, quorumNumbers []byte) (cstaskmanager.IAlignedLayerTaskManagerTask, uint32, error) {
```
Comment (entropidelic) [inline core/chainio/avs_writer.go]: this could be removed now that that `SendTask` is implemented.

**Comment by entropidelic** [comment]
PR looks good, just a little comment that IMO would make the code a little simpler and more legible. 
There is a function `sendTask` in `task_sender/cmd/main.go`. To avoid repeating the name too much, it could be better to rename it something like `taskSenderMain`.

**Comment by JuArce** [comment]
> PR looks good, just a little comment that IMO would make the code a little simpler and more legible. There is a function `sendTask` in `task_sender/cmd/main.go`. To avoid repeating the name too much, it could be better to rename it something like `taskSenderMain`.

I agree. As CLI receives a parameter called Action
`Action:      sendTask`
I can rename `sendTask` to `sendTaskAction`

**Review by entropidelic — APPROVED** [review]

File: contracts/src/core/AlignedLayerServiceManager.sol [hunk 2]
```diff
@@ -19,6 +19,21 @@ import {IStakeRegistry} from "eigenlayer-middleware/interfaces/IStakeRegistry.so
...
+    struct Task {
+        uint16 verificationSystemId;
+        bytes proof;
+        bytes pubInput;
+        uint32 taskCreatedBlock;
+    }
+
+    /* STORAGE */
+    // The latest task index
+    uint32 public latestTaskNum;
```
Comment (MauroToscano) [inline contracts/src/core/AlignedLayerServiceManager.sol]: We should make this number bigger than an uint32

File: core/chainio/avs_writer.go [hunk 3]
```diff
@@ -82,26 +87,40 @@ func NewAvsWriterFromConfig() (*AvsWriter, error) {
...
+	)
+	if err != nil {
+		w.logger.Error("Error assembling CreateNewTask tx", "err", err)
+		return servicemanager.AlignedLayerServiceManagerTask{}, 0, err
+	}
+	// TODO wait for transaction receipt. ethClient does not have this method
+	// EigenSDK has a method called WaitForTransactionReceipt in InstrumentedEthClient
+	// But is needs telemetry to work
+	// https://github.com/Layr-Labs/eigensdk-go/blob/master/chainio/clients/eth/instrumented_client.go
+	//receipt := avsWriter.Client.WaitForTransactionReceipt(context.Background(), tx.Hash())
```
Comment (MauroToscano) [inline core/chainio/avs_writer.go:74]: Create an issue for this

File: contracts/script/output/devnet/alignedlayer_deployment_output.json [hunk 4]
```diff
@@ -26,4 +26,4 @@
     "alignedLayerUpgrader": "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
     "pauserRegistry": "0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512"
   }
-}
+}
```
Comment (MauroToscano) [inline contracts/script/output/devnet/alignedlayer_deployment_output.json]: Add newline

**Review by MauroToscano — CHANGES_REQUESTED** [review]

**Comment by MauroToscano** [comment]
This closes #15, right ?

**Comment by MauroToscano** [comment]
This closes #19, right ?

**Comment by JuArce** [comment]
> This closes #19, right ?

Yes. The issue is linked to the pr

File: contracts/script/output/devnet/alignedlayer_deployment_output.json (hunk 4 above)
Comment (JuArce) [reply contracts/script/output/devnet/alignedlayer_deployment_output.json]: Every time the file is generated, there is missing the newline

File: core/chainio/avs_writer.go (hunk 3 above)
Comment (MauroToscano) [reply core/chainio/avs_writer.go:74]: Solved

File: contracts/script/output/devnet/alignedlayer_deployment_output.json (hunk 4 above)
Comment (MauroToscano) [reply contracts/script/output/devnet/alignedlayer_deployment_output.json]: Maybe we should change it to add the newline

Comment (MauroToscano) [inline contracts/script/output/devnet/alignedlayer_deployment_output.json]:
```suggestion
}

```

File: contracts/src/core/AlignedLayerServiceManager.sol [hunk 5]
```diff
@@ -19,6 +19,21 @@ import {IStakeRegistry} from "eigenlayer-middleware/interfaces/IStakeRegistry.so
...
+    // EVENTS
+    event NewTaskCreated(uint32 indexed taskIndex, Task task);
+
+    // STRUCTS
+    struct Task {
+        uint16 verificationSystemId;
+        bytes proof;
+        bytes pubInput;
+        uint32 taskCreatedBlock;
+    }
```
Comment (MauroToscano) [inline contracts/src/core/AlignedLayerServiceManager.sol:31]: Is the block needed ? Why is it there ? Also the fields are not correct but we can change it

**Comment by MauroToscano** [comment]
> > This closes #19, right ?
> 
> Yes. The issue is linked to the pr

Oh I see, usually I just write closes and the number of the PR so it gets closed. But I see it's linked

**Review by MauroToscano — APPROVED** [review]

**Pull request merged**